File: backend/routes/user_routes.py
from flask import Blueprint, jsonify, request
from controller.user_controller import UserController

user_routes = Blueprint('user_routes', __name__)

@user_routes.route('/users', methods=['GET'])
def get_users():
    users = UserController.get_all_users()
    return jsonify(users)

@user_routes.route('/users/<int:user_id>', methods=['GET'])
def get_user(user_id):
    user = UserController.get_user(user_id)
    return jsonify(user)

@user_routes.route('/users', methods=['POST'])
def create_user():
    user_data = request.json
    result = UserController.create_user(user_data)
    return jsonify(result), 201

File: backend/routes/userRoutes.js
const express = require('express');
const router = express.Router();
const {
    getUsers,
    getUser,
    createUser,
    updateUser,
    deleteUser,
    loginUser
} = require('../controllers/userController');

// Login user
router.post('/login', loginUser);

// Get all users
router.get('/', getUsers);

// Get a single user
router.get('/:userId', getUser);

// Create a new user
router.post('/', createUser);

// Update a user
router.put('/:userId', updateUser);

// Delete a user
router.delete('/:userId', deleteUser);


module.exports = router;
